### PR TITLE
Ensure tests skipped due to associated compile failure get added to test_changes.list; group rt log failure messages by failure reason

### LINKED ISSUE #49: WCOSS2 Testing PR #3313
Tacking my routine testing for UFSWM on WCOSS2:

*blocks ufs-community/ufs-weather-model/pull/3313

### LINKED ISSUE #3018: test_changes.list does not report tests that failed bc their compile failed
## Description
When the COMPILE step fails for a set of RTs, those RTs are listed in the log as SKIPPED, and they do not show up in `test_changes.list` as failed tests. 

### To Reproduce:
1. Run RTs where the compile step fails (this happens spontaneously but could probably be simulated by intentionally adding a typo to the code). 
2. See that the log shows a compile failure and lists associated tests as skipped. 
3. Open test_changes.list, and note that the skipped tests are not listed. 

## Additional context
For example, see [this commit](https://github.com/ufs-community/ufs-weather-model/pull/2983/files/f4ee8f3b341c8abe79fe148941ab32b73c8c5c6d#diff-cb5888ef341590aa3365429eb6ce948d2e67b1425fd94f2ebd0b354833f81af0).

### PR DESCRIPTION
**Pull request by gspetro-NOAA**
## Commit Queue Requirements:
- [X] This PR addresses a relevant WM issue (if not, create an issue). 
- [X] All subcomponent pull requests (if any) have been reviewed by their code managers. **N/A**
- [x] Run the full Intel+GNU RT suite (compared to current baselines), preferably on Ursa (Derecho or Hercules are acceptable alternatives). **Exceptions:** documentation-only PRs, CI-only PRs, etc.
   - [x] Commit log file w/full results from RT suite run (if applicable).
   - [x] Verify that `test_changes.list` indicates which tests, if any, are changed by this PR. Commit `test_changes.list`, even if it is empty.
- [X] Transparency in the use of generative AI is required by NOAA policy. Was GenAI used in this work? Just asked Gemini some questions!
   - [ ] [Generative AI tool NOAA Gemini] was used to assist with developing this code. The code has been reviewed, edited, and validated by NWS staff.
- [x] Fill out all sections of this template.

---

## Description:
When a compile fails, the tests that are skipped are not added to `test_changes.list`; this PR fixes that. Additionally, this PR renames the `-b` option to `-s` (for subset) because it can be used to run _any_ subset of RTs formatted like `test_changes.list` (e.g., `"test_name compiler"`). The `-b` option implies that it can only be used to create a new baseline for a subset of tests in combination with the `-c` option. 

In addition to running the full RT suite, I tested this PR by: 
- Altering `UFSATM/fv3/atmos_cubed_sphere/model/fv_dynamics.F90` with a line ("THIS_IS_AN_INTENTIONAL_COMPILATION_ERROR !!!") to cause a compile fail. 
- Running `nohup ./rt.sh -a epic -e -s run.list &` with the following `run.list` file: 
  ```
  cpld_warmstart_c12_9deg intel
  hafs_global_storm_following_1nest_atm intel
  conus13km_2threads gnu
  control_diag_debug gnu
  control_restart_p8 intel
  datm_cdeps_control_cfsr intelllvm
  cpld_control_gefs intel
  cpld_restart_gefs intel
  cpld_dcp_gefs intel
  ```

Only the datm test passed, which makes sense given the lack of active atm component. All of the other tests, plus `control_p8 intel` (needed for `control_restart_p8 intel`) appear in `test_changes.list`. 

UPDATES: 
This PR also reworks rt.sh to group failed/skipped tests by failure/skip reason. For example, see: 
[RegressionTests_ursa_3313a.log](https://github.com/user-attachments/files/30992812/RegressionTests_ursa_3313a.log).

### Commit Message:

```
* UFSWM - Bugfix to ensure tests skipped due to associated compile failure get added to test_changes.list
* UFSWM - Group RT log messages for failed/skipped tests by failure/skip reason
```

### Priority:
- [ ] Critical Bugfix: Reason
- [ ] High: Reason
- [X] Normal

### UFSWM:
* Closes #3018 

### Sub component Pull Requests:
* None

### UFSWM Blocking Dependencies:

- [ ] Blocked by #
- [X] None

### Documentation:

- [ ] Documentation update required. 
   - [ ] Relevant updates are included with this PR. 
   - [ ] A WM issue has been opened to track the need for a documentation update; a person responsible for submitting the update has been assigned to the issue (link issue).
- [X] Documentation update NOT required. 
   - Explanation: 

---
### Regression Test Changes (Please commit test_changes.list):

- [ ] PR Adds New Tests/Baselines.
- [ ] PR Updates/Changes Baselines.
- [X] No Baseline Changes.

### Input data Changes:

- [X] None.
- [ ] PR adds input data.
- [ ] PR changes existing input data.

### Library Changes/Upgrades:

- [ ] Required
  * Library names w/versions:
  * Git Stack Issue (JCSDA/spack-stack#)
- [X] No Updates
  
---

## Testing Log:
- RDHPCS
  - [ ] Orion
  - [ ] Hercules
  - [ ] GaeaC6
  - [ ] Derecho
  - [ ] Ursa
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- [ ] CI
- [ ] opnReqTest (complete task if unnecessary)

## Testing Remarks:

-

### CODE REVIEW

**Comment by DeniseWorthen** [comment]
@gspetro-NOAA Maybe I'm misunderstanding, but I consider ``test_changes.list`` the list of tests whose baselines change. If the PR code doesn't even compile for some test(s), then we don't want to associate 'failed compile' with 'baseline change'. 

The PR with a failed compile simply needs fixing before any testing should proceed. I'd like to see the RT log certainly spell this out much more boldly; it is easy to miss. But hiding a failed compile in the ``test_changes.list`` doesn't really solve the problem of someone making a PR, expecting some tests to fail anyway and missing that some tests actually failed because they didn't compile.

**Comment by gspetro-NOAA** [comment]
> @gspetro-NOAA Maybe I'm misunderstanding, but I consider `test_changes.list` the list of tests whose baselines change. If the PR code doesn't even compile for some test(s), then we don't want to associate 'failed compile' with 'baseline change'.
> 
> The PR with a failed compile simply needs fixing before any testing should proceed. I'd like to see the RT log certainly spell this out much more boldly; it is easy to miss. But hiding a failed compile in the `test_changes.list` doesn't really solve the problem of someone making a PR, expecting some tests to fail anyway and missing that some tests actually failed because they didn't compile.

@DeniseWorthen I agree that the _committed_ `test_changes.list` file should only contain tests that failed due to a baseline comparison error because they are expected to change baselines. Currently, however, a test may end up listed in `test_changes.list` because it failed for other non-PR-related reasons during testing, such as a disk quota issue or test time out. Realistically, the developer always needs to check the logs to see why things failed and if they need fixing. This is true whether the test failed in compile or failed during testing with, e.g., FAILED: RUN DID NOT COMPLETE. And whether a rerun or a fix-and-rerun is required, the developer (or CM) should ideally be able to rerun all tests that didn't pass, whether they were skipped due to compile failure or failed mid-test. To my knowledge, we don't have any functionality that does this, so we end up looking in `rt.conf` and copy-pasting test names and compilers from under the failed compiles. If there's already a better way, I'd love to know! This was particularly problematic with Derecho testing  (although now things are okay there other than the super long queue times). 

I think things are getting confused because the -b option is somewhat misleadingly named to imply that it only applies to baseline generation. Technically, running with the `-b` option (and no `-c`) will work to run a subset of tests, and the file it takes in _could_ be named anything. So `-b` is **not** exclusively used for baseline generation of a subset of tests. Similarly, the file produced by an RT run will be called `test_changes.list` regardless of whether test baselines change. We could rename it `failed_tests.list`, but I like the idea of keeping the current name because when used locally, the name doesn't matter much, and when pushed to the PR, it does (or should) reflect the tests whose baselines are expected to change.

**Comment by DeniseWorthen** [comment]
I went back and looked at the original PR that led to the issue creation; the PR in that case was made by one of our most careful contributors and even then the "noise" of a compile failure didn't break through. 

I pasted the end of the log file from that PR into an AI prompt "What messaging would you suggest to clearly announce loudly and boldly that there were failures in addition to just baseline changes."

The suggestion was to cut through the noise by delineating the message types:
Screenshot 2026-08-07 at 6 42 41 AM


While I don't disagree in principle that adding a compile fail to a test_changes.list could be useful (as long as it was not committed), I think the root of the problem that led to the original issue formation is the messaging, not the test_changes.list

**Comment by gspetro-NOAA** [comment]
The messaging is definitely big part of the challenge. We've run into the issue in other PRs, where there were a lot of expected baseline changes and one/a few other little failures snuck through. PR #3130 comes to mind. As a result, I put in [some CI](https://github.com/ufs-community/ufs-weather-model/actions/runs/31144523040?pr=3313) that parses out which tests failed for what reasons, but it's combined with the warnings/remarks check and doesn't produce an actual failure (because sometimes we expect tests to fail). It could be improved, but it would probably be more useful to have the failure messaging in the logs so no one has to go looking for it. I will see about updating the logs to print failures category by category as the AI suggested... 🤔 
With this potential addition, are you ok with the changes that are already in?

**Comment by DeniseWorthen** [comment]
>  Realistically, the developer always needs to check the logs to see why things failed and if they need fixing. This is true whether the test failed in compile or failed during testing with, e.g., FAILED: RUN DID NOT COMPLETE. And whether a rerun or a fix-and-rerun is required, the developer (or CM) should ideally be able to rerun all tests that didn't pass, whether they were skipped due to compile failure or failed mid-test. To my knowledge, we don't have any functionality that does this, so we end up looking in `rt.conf` and copy-pasting test names and compilers from under the failed compiles. If there's already a better way, I'd love to know! This was particularly problematic with Derecho testing (although now things are okay there other than the super long queue times).

I do agree that manually editing the rt.conf for failures is not ideal, and if that is the primary purpose here, I think that is OK as far as it goes. I would love to see better log messaging though.

File: tests/rt.sh [hunk 1]
```diff
@@ -14,7 +14,6 @@ usage() {
   echo "Usage: $0 -a <account> | -b <file> | -c | -d | -e | -h | -k | -l <file> | -m | -n <name> | -o | -r | -v | -w | -x"
```
Comment (BrianCurtis-NOAA) [inline tests/rt.sh]: i miss this a lot, you'll want to move the -b to -s here as well.

**Review by BrianCurtis-NOAA — CHANGES_REQUESTED** [review]

File: tests/rt.sh (hunk 1 above)
Comment (gspetro-NOAA) [reply tests/rt.sh]: Nice catch! Thanks!

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -16,8 +16,10 @@
 - [ ] Run the full Intel+GNU RT suite (compared to current baselines), preferably on Ursa (Derecho or Hercules are acceptable alternatives). **Exceptions:** documentation-only PRs, CI-only PRs, etc.
    - [ ] Commit log file w/full results from RT suite run (if applicable).
    - [ ] Verify that `test_changes.list` indicates which tests, if any, are changed by this PR. Commit `test_changes.list`, even if it is empty.
-- [ ] Transparency in the use of generative AI is required by NOAA policy. Was GenAI used in this work?
-   - [ ] [Generative AI tool (insert name, if any)] was used to assist with developing this code. The code has been reviewed, edited, and validated by NWS staff.
+- Transparency in the use of generative AI is required by NOAA policy. Was GenAI used in this work?
+   - [ ] No.
+   - [ ] Yes. [Generative AI tool (insert name, if any)] was used to assist with developing this code. 
+     - [ ] The code has been reviewed, edited, and validated by NWS staff.
```
Comment (gspetro-NOAA) [inline .github/pull_request_template.md:22]: @BrianCurtis-NOAA @NickSzapiro-NOAA @dpsarmie People have just been not checking off the AI boxes at all. I was thinking for making things more explicit where they have to check a yes or no. How does this look?

Comment (dpsarmie) [reply .github/pull_request_template.md:22]: That looks good to me.

Comment (gspetro-NOAA) [reply .github/pull_request_template.md:22]: I'm also wondering... how does code get "reviewed, edited, and validated by NWS staff"? Is the idea that we need an NWS person to approve the PR? If I don't know, regular devs surely won't either. 😜

Comment (NickSzapiro-NOAA) [reply .github/pull_request_template.md:22]: Sure

Mostly, this is the language we were told to use. 
But, OMD CMs are also NWS (contract) staff and do these for AI and non-AI PRs.

**Comment by DeniseWorthen** [comment]
@gspetro-NOAA It looks like your effort to group the failures does not actually separate them visually into blocks of text. What you're proposing is better than the current, but I think it could still be better. Is there a reason you decided not to visually block them (blank lines) and call out the failures in red?

**Comment by gspetro-NOAA** [comment]
> @gspetro-NOAA It looks like your effort to group the failures does not actually separate them visually into blocks of text. What you're proposing is better than the current, but I think it could still be better. Is there a reason you decided not to visually block them (blank lines) and call out the failures in red?

@DeniseWorthen I can definitely add in a blank line! Easy enough. As to calling out the failures in red, I'm not sure how I can in a plain text file because they don't support markup/styling. Wouldn't it depend on people's editors/terminals and what color scheme they have set up?

File: .github/pull_request_template.md (hunk 2 above)
Comment (gspetro-NOAA) [reply .github/pull_request_template.md:22]: Great! Ok, just wanted to make sure we're not having to call on people outside the current CM team, bc that could complicate/prolong things.

**Comment by DeniseWorthen** [comment]
@gspetro-NOAA I can't find the AI history where I generated the sample w/ the red texts; but I specifically remember being asked whether people would be viewing in terminals in reference to the colors. But since I can't reproduce it now, I guess we'll have to see how much grouping/separating helps. Your solution is an improvement at least.

**Comment by gspetro-NOAA** [comment]
> @gspetro-NOAA I can't find the AI history where I generated the sample w/ the red texts; but I specifically remember being asked whether people would be viewing in terminals in reference to the colors. But since I can't reproduce it now, I guess we'll have to see how much grouping/separating helps. Your solution is an improvement at least.

@DeniseWorthen I did find that it's possible to embed ANSI escape sequences to add color when viewing the logs in the terminal. The only concern is that if someone is ever NOT viewing the logs in the terminal, there will be a bunch of random characters added that would be rendered as color in the terminal but not elsewhere. I do think most people will be viewing in the terminal, but maybe not 100% of the time, so not sure the best path forward on that... 🤔

**Comment by gspetro-NOAA** [comment]
@DeniseWorthen After some experimentation, I found that I can style the output, but depending on how it's being viewed, it can be either very helpful or somewhat unreadable. The ANSI escape sequences don't render properly if I do, e.g., `vi logs/RegressionTests_ursa.log`: 

image


However, if I run `cat logs/RegressionTests_ursa.log` instead, it's great! 

image

Note that if we go this route, I would change "Failed Tests" and Failed Compiles" to the same color--this isn't the final product; just some experimentation w/what's possible! 

So I guess the question is, now what? I can think of a few options, tho I'm open to more if you think of something else: 
1. Style w/whitespace/indentation only
2. Style w/ANSI escape sequences. Provide a script to strip the sequences if desired. Provide instructions (where???) on which commands will/won't work to view colorized/stylized version. 
3. Provide a command line option to style the logs w/ANSI escape sequences. Default would be to style w/whitespace only, not ANSI escape sequences.

**Comment by DeniseWorthen** [comment]
@gspetro-NOAA Thanks for trying to make this work. I think just colors are going to end up more trouble than they're worth (?). Maybe we should just stick to calling out w/ spacing, *** MESSAGE HERE *** (or something like that) and see how much that helps.

**Comment by BrianCurtis-NOAA** [comment]
I wonder if we can get rid of all the paths and just put it on there once:
(I feel the logpath on each one makes it too verbose)

```
LOGPATH: /scratch3/NAGAPE/epic/Gillian.Petro/ursa/RTs/ufs-wm/pretests/rtsh/tests/logs/log_ursa/

UNABLE TO FINISH **COMPILE**:
* s2sw_pdlib_intel

FAILED TESTS
ASSOCIATED COMPILE FAILED:
* cpld_warmstart_c12_9deg_intel
* cpld_restart_c12_9deg_intel

**RUN** DID NOT COMPLETE:
* datm_cdeps_control_cfsr_intel
* datm_cdeps_control_cfsr_intelllvm
```

**Comment by gspetro-NOAA** [comment]
@DeniseWorthen @BrianCurtis-NOAA Thanks for the feedback! Here are the latest versions. Let me know if you prefer the first or second and if there are any additional changes you'd make! 

Option 1:
image

Option 2:
image

**Comment by BrianCurtis-NOAA** [comment]
If it's easy to move sections around, it might be worth having which compiles outright failed to complete first (immediate action), then move to tests that failed to run (or complete) second, then tests that failed comparison (baseline changes), then tests that were skipped due to compile failure.

**Comment by DeniseWorthen** [comment]
I'd probably prefer option 1, a little 'blockier'---the three stars delineating the blocks.

Getting picky....I'd prefer to see the "unable to start test" listed last; it is the least informative in my mind since we should know it didn't complete because the control test associated w/ it didn't run.
 
 So in order of most alarming:
Screenshot 2026-08-19 at 4 04 12 PM

Screenshot 2026-08-19 at 4 04 17 PM

**Comment by gspetro-NOAA** [comment]
> If it's easy to move sections around, it might be worth having which compiles outright failed to complete first (immediate action), then move to tests that failed to run (or complete) second, then tests that failed comparison (baseline changes), then tests that were skipped due to compile failure.

At the moment (after some changes), I think I basically have that order, but it's very easy to switch around. Does this work in terms of order?

Failed Compiles: 
- UNABLE_TO_START_COMPILE
- UNABLE_TO_FINISH_COMPILE
- COMPILE_DISK_QUOTA_ISSUE
- COMPILE_TIMED_OUT

Failed Tests:
- RUN_DID_NOT_COMPLETE
- TEST_DISK_QUOTA_ISSUE
- TEST_TIMED_OUT
- UNABLE_TO_START_TEST
- UNSUCCESSFUL_BASELINE_COMPARISON
- UNABLE_TO_COMPLETE_COMPARISON
- DOES_NOT_GENERATE_BASELINE
- ASSOCIATED_COMPILE_FAILED

**Comment by gspetro-NOAA** [comment]
A couple of log samples with the latest iteration: 

image


Another example: 

image

I am questioning whether we want the line about `*** # NON_BASELINE FAILURES -- NOT just baseline changes ***` because at the moment, it's not counting tests skipped (whether for compile failure or tests that are _supposed_ to be skipped because they don't generate a baseline)? The number shown and the number of tests actually failing are different, and not just because the baseline failures aren't counted. Maybe I should just put `*** WARNING -- not just baseline change failures ***` or something similar rather than calculating an exact number? I can also just change the calculation.

**Comment by DeniseWorthen** [comment]
I think it is fine not to have a count; I'm not really sure you need the line at all. 

Of your two latest, the lower one has "unable to complete comparison". I've always felt that was so ambiguous. It reads as though it started to compare, but couldn't finish. Was it unable because it didn't compile, didn't run, didn't complete or what? 

I'm pretty sure it actually means "the baseline comparisons failed"; we should just say so.

**Comment by gspetro-NOAA** [comment]
> I think it is fine not to have a count; I'm not really sure you need the line at all.
> 
> Of your two latest, the lower one has "unable to complete comparison". I've always felt that was so ambiguous. It reads as though it started to compare, but couldn't finish. Was it unable because it didn't compile, didn't run, didn't complete or what?
> 
> I'm pretty sure it actually means "the baseline comparisons failed"; we should just say so.

I can definitely do that, but one confusing (to me) thing I came across in rt.sh is that there are [two, seemingly related statuses](https://github.com/ufs-community/ufs-weather-model/blob/a5fb3be70c056126ca6e997405ebf4877234938f/tests/rt.sh#L350-L360): UNABLE TO COMPLETE COMPARISON and UNSUCCESSFUL BASELINE COMPARISON. Do you know when UNSUCCESSFUL BASELINE COMPARISON would be relevant? In theory, it seems like UNABLE TO COMPLETE COMPARISON would be when there is a new test added, and no baseline exists, so there is nothing to compare against. Then UNSUCCESSFUL BASELINE COMPARISON would be when the baseline exists but is not identical. However, in practice, I think both of those situations end up with UNABLE TO COMPLETE COMPARISON in the logs. Or am I imagining that?

**Comment by DeniseWorthen** [comment]
I can't be 100% sure, about those two messages. I'm not sure I've seen UNSUCCESSFUL BASELINE COMPARISON--can you recall seeing it? I do definitely see the UNABLE TO COMPLETE message. 

Brian might know better, since I think he added all the messaging at some point.

**Comment by gspetro-NOAA** [comment]
> I can't be 100% sure, about those two messages. I'm not sure I've seen UNSUCCESSFUL BASELINE COMPARISON--can you recall seeing it? I do definitely see the UNABLE TO COMPLETE message.
> 
> Brian might know better, since I think he added all the messaging at some point.

I don't think I've ever seen UNSUCCESSFUL BASELINE COMPARISON. Hopefully @BrianCurtis-NOAA can enlighten us when he's back from PTO... Not sure that rt.sh ever gets to that block in the conditional evaluation, but maybe once upon a time it did or there's an edge case where it still does?

**Comment by BrianCurtis-NOAA** [comment]
> > I think it is fine not to have a count; I'm not really sure you need the line at all.
> > Of your two latest, the lower one has "unable to complete comparison". I've always felt that was so ambiguous. It reads as though it started to compare, but couldn't finish. Was it unable because it didn't compile, didn't run, didn't complete or what?
> > I'm pretty sure it actually means "the baseline comparisons failed"; we should just say so.
> 
> I can definitely do that, but one confusing (to me) thing I came across in rt.sh is that there are [two, seemingly related statuses](https://github.com/ufs-community/ufs-weather-model/blob/a5fb3be70c056126ca6e997405ebf4877234938f/tests/rt.sh#L350-L360): UNABLE TO COMPLETE COMPARISON and UNSUCCESSFUL BASELINE COMPARISON. Do you know when UNSUCCESSFUL BASELINE COMPARISON would be relevant? In theory, it seems like UNABLE TO COMPLETE COMPARISON would be when there is a new test added, and no baseline exists, so there is nothing to compare against. Then UNSUCCESSFUL BASELINE COMPARISON would be when the baseline exists but is not identical. However, in practice, I think both of those situations end up with UNABLE TO COMPLETE COMPARISON in the logs. Or am I imagining that?

This may not be caught because of a typo (read: brain fart) when putting this all together in my head. As seen here: https://github.com/ufs-community/ufs-weather-model/blob/a5fb3be70c056126ca6e997405ebf4877234938f/tests/rt.sh#L349 I think the compiles never output an rt_... file, so this logic would never be entered.

**Comment by gspetro-NOAA** [comment]
> This may not be caught because of a typo (read: brain fart) when putting this all together in my head. As seen here:
> 
> https://github.com/ufs-community/ufs-weather-model/blob/a5fb3be70c056126ca6e997405ebf4877234938f/tests/rt.sh#L349
> 
> I think the compiles never output an rt_... file, so this logic would never be entered.

The compiles don't, but the tests sometimes do. On the latest run, I see these `rt_*` files: 
```
rt_rap_control_intel.log
rt_catchem_control_intelllvm.log
rt_datm_cdeps_control_cfsr_intel.log
rt_datm_cdeps_control_cfsr_intelllvm.log
rt_datm_cdeps_gfs_intel.log
rt_datm_cdeps_restart_cfsr_intel.log
```
I've always wondered what they were for but never needed to check them. It looks like they literally just list the files being compared. For example: 

```
Checking test rap_control_intel results ....
 Comparing sfcf000.nc .....USING NCCMP......OK
 Comparing sfcf009.nc .....USING NCCMP......OK
 Comparing sfcf012.nc .....USING NCCMP......OK
 Comparing atmf000.nc .....USING NCCMP......OK
 Comparing atmf009.nc .....USING NCCMP......OK
```

Versus:
```
Checking test catchem_control_intelllvm results ....
 Comparing sfcf000.nc ............MISSING baseline
 Comparing sfcf024.nc ............MISSING baseline
 Comparing atmf000.nc ............MISSING baseline
 Comparing atmf024.nc ............MISSING baseline
 Comparing GFSFLX.GrbF00 ............MISSING baseline
 Comparing GFSFLX.GrbF24 ............MISSING baseline
```

For reference, this is the RegressionTests_ursa.log output (I had deleted the catchem baseline from my created baselines to generate the status):

```
...
BASELINE DIRECTORY: /scratch4/NAGAPE/epic/Gillian.Petro/stmp/RT_BASELINE/Gillian.Petro/FV3_RT/REGRESSION_TEST
COMPARISON DIRECTORY: /scratch4/NAGAPE/epic/Gillian.Petro/stmp/RT_RUNDIRS/Gillian.Petro/FV3_RT/rt_3542568

RT.SH OPTIONS USED:
* (-a) - HPC PROJECT ACCOUNT: epic
* (-s) - RUN TESTS FROM FILE: short.list
* (-m) - COMPARE AGAINST CREATED BASELINES
* (-e) - USE ECFLOW

PASS -- COMPILE 's2sw_pdlib_intel' [17:11, 15:40] ( 1033 remarks )
FAILED: RUN DID NOT COMPLETE -- TEST 'cpld_warmstart_c12_9deg_intel' [, ]( MB)
FAILED: UNABLE TO START TEST -- TEST 'cpld_restart_c12_9deg_intel' [, ]( MB)

PASS -- COMPILE 'atm_dyn32_intel' [07:11, 05:53] ( 1 warnings 523 remarks )
FAILED: TEST TIMED OUT -- TEST 'control_p8_intel' [, ]( MB)
FAILED: UNABLE TO START TEST -- TEST 'control_restart_p8_intel' [, ]( MB)

PASS -- COMPILE 'rrfs_intel' [07:11, 05:31] ( 3 warnings 450 remarks )
PASS -- TEST 'rap_control_intel' [05:31, 02:31](1769 MB)

PASS -- COMPILE 'datm_cdeps_intel' [14:11, 12:12] ( 573 remarks )
PASS -- TEST 'datm_cdeps_control_cfsr_intel' [04:18, 02:00](1923 MB)
PASS -- TEST 'datm_cdeps_restart_cfsr_intel' [03:17, 01:17](1872 MB)
PASS -- TEST 'datm_cdeps_gfs_intel' [09:18, 06:25](4789 MB)

PASS -- COMPILE 'datm_cdeps_intelllvm' [14:11, 12:44]
PASS -- TEST 'datm_cdeps_control_cfsr_intelllvm' [04:18, 01:56](1926 MB)

PASS -- COMPILE 'catchem_intelllvm' [11:11, 09:23]
FAILED: UNABLE TO COMPLETE COMPARISON -- TEST 'catchem_control_intelllvm' [, ]( MB)

SYNOPSIS:
Starting Date/Time: 20260825 16:22:38
Ending Date/Time: 20260825 17:30:06
Total Time: 01h:07m:37s
Compiles Completed: 6/6
Tests Completed: 5/10

LOGPATH: /scratch3/NAGAPE/epic/Gillian.Petro/ursa/RTs/ufs-wm/pretests/3313-rtsh2/tests/logs/log_ursa
  * Compile logs located at: compile_<compile_name>_<compiler>.log
  * Test logs located at: run_<test_name>_<compiler>.log


*** FAILED TESTS ***

RUN DID NOT COMPLETE: 

  * cpld_warmstart_c12_9deg intel
  * control_p8 intel

TEST TIMED OUT: 

  * control_p8 intel

UNABLE TO START TEST: 

  * cpld_restart_c12_9deg intel
  * control_restart_p8 intel

UNABLE TO COMPLETE COMPARISON: 

  * catchem_control intelllvm


NOTES:
A file 'test_changes.list' was generated with list of all failed tests.
...
```
So it seems like if there's a `fail_test_*` file and an `rt_*` log file, we may need to grep the `rt_*` log file to see if it's got MISSING BASELINE or just NOT OK if we care about the distinction between UNABLE TO COMPLETE COMPARISON and UNSUCCESSFUL BASELINE COMPARISON.

Another thing to consider is what to do when multiple statuses are true. For example, in this sample log, control_p8_intel is printed under RUN DID NOT COMPLETE and TEST TIMED OUT. The run didn't complete _because_ the test timed out, so it's not wrong, but do we want both statuses reported...?

**Comment by DeniseWorthen** [comment]
This gets very convoluted, doesn't it? 

n my own mind "run did not complete" would indicate the run actually died---seg fault, mis-configuration (like forgetting to turn waves to false in MOM_input so it dies). Timed out is just that---the run was doing fine, it just ran out of time.

**Comment by BrianCurtis-NOAA** [comment]
I'm fine if we want to re-imagine how this part is done.

**Comment by gspetro-NOAA** [comment]
Reorganizing the conditional blocks as we speak! 😄

**Comment by gspetro-NOAA** [comment]
Ok, now I've got this log! But it would be good if I could get a second pair of eyes on the changes in that section of `rt.sh` to make sure I'm not missing edge cases or failing to consider some important case that I haven't tested for. In theory, tests should only have one fail status now. UNSUCCESSFUL BASELINE COMPARISON is when files are different, and UNABLE TO COMPLETE COMPARISON is when the baseline does not exist. I'm going to rename those statuses, but I wanted to test one thing at a time and make sure the logic works first before renaming things. 

image

**Comment by DeniseWorthen** [comment]
@gspetro-NOAA I looked at your RT run directory and also the logs; this seems pretty good to me!

 - the warmstart test actually fails, so 'run did not complete' is correct. 
 - control_p8 shows timed-out in err file
 - datm doesn't reproduce baseline 
 - catchem has a missing baseline
 
 I think renaming those last two status messages is the right move.

**Comment by gspetro-NOAA** [comment]
The latest: 

image

**Comment by gspetro-NOAA** [comment]
@BrianCurtis-NOAA @DeniseWorthen Do you all have additional change requests? Or specific testing you'd like to see? If not, I can go ahead and run the full RT suite/pretesting, but I'm open to more suggestions if you have them! I did address the change that Brian requested, so I think the change request can be removed (unless there are additional ideas)? 

Also, here's another example with some compile failures: 

image

**Comment by BrianCurtis-NOAA** [comment]
Maybe: "ASSOCIATED COMPILE FAILED: TESTS NOT RUN" ?

**Comment by gspetro-NOAA** [comment]
> Maybe: "ASSOCIATED COMPILE FAILED: TESTS NOT RUN" ?

Voilà!

image

**Comment by gspetro-NOAA** [comment]
Any more suggestions for me? 😁

**Comment by BrianCurtis-NOAA** [comment]
I appreciate your patience with my many small changes. I'm looking forward to seeing these changes live. Nothing more from me.

**Comment by DeniseWorthen** [comment]
@gspetro-NOAA Looks good to me. Certainly an improvement. Thanks for all working through all the changes.

**Review by gspetro-NOAA — COMMENTED** [review]
All tests passed successfully. Skipping Derecho due to ongoing maintenance and Acorn due to ongoing unavailability.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by FernandoAndrade-NOAA — APPROVED** [review]
